### DISCUSSION TITLE
igl::offset_surface

### DISCUSSION THREAD

**Discussion by 3Dkimbob** [libigl/libigl#1934] [Q&A] [answered]
The result is strange when offset surface is used in the bounded mesh.

The voxel grid seems to be working strangely on the boundary side.

How should I use this to make the border look normal?

Using code
```
#include "NativeIGL.h"

#include <igl/signed_distance.h>
#include <igl/offset_surface.h>
#include <time.h>


NativeMesh NativeIGL_OffsetMesh(static NativeMesh mesh,int s,double isolevel)
{
    Eigen::MatrixXd V, SV;
    Eigen::MatrixXi F, SF;
    Eigen::MatrixXd GV;
    Eigen::VectorXd S;
    Eigen::RowVector3i side;

    isolevel = isolevel < 0 ? 0 : isolevel;
    s = s < 0 ? 0 : s;
    //igl::SignedDistanceType signed_distance_type = igl::SignedDistanceType::SIGNED_DISTANCE_TYPE_DEFAULT;
    igl::SignedDistanceType signed_distance_type = igl::SignedDistanceType::SIGNED_DISTANCE_TYPE_PSEUDONORMAL;
    std::cout << "Start Offset Surface - isoLevel: " << isolevel << ", s: " << s << std::endl;
    clock_t startTime = clock();
    ConvertMesh2Eigen(mesh, V, F);
    igl::offset_surface(V, F, isolevel, s, signed_distance_type, SV, SF, GV, side, S);
    std::cout << "End offset surface: " << clock() - startTime << std::endl;
    return GenerateMeshFromEigen(SV, SF);
}
```

-original
![image](https://user-images.githubusercontent.com/38707850/145208994-b6456740-5e40-4e24-aa6e-2caeaa91fb9b.png)
![image](https://user-images.githubusercontent.com/38707850/145209026-64a17bcc-a5de-4dd4-9096-69b5c70336c5.png)

-result
![image](https://user-images.githubusercontent.com/38707850/145208932-0760ddd2-03e3-4f60-b439-7c5612aab03c.png)
![image](https://user-images.githubusercontent.com/38707850/145209068-cf8ca2d4-7c37-4e0e-90c1-607b0e4413c6.png)

**Comment by alecjacobson** [comment, marked as answer]
Try changing `SIGNED_DISTANCE_TYPE_PSEUDONORMAL` to `
SIGNED_DISTANCE_TYPE_FAST_WINDING_NUMBER`

**Reply by 3Dkimbob** [reply]
Thank you for helping me.

I tested it while changing the signed distance type (SDT), resolution, and internal code.
Since SDT is PSEUDONORMAL and FloodFill is removed, it seems to be working as desired. Additionally, it seems possible to get the mesh you want by adding only the method of removing Nanvertex.

Is the way I'm going to do it right?

(If the meaning is weird, it's because it's a translator.)

"nan vertex having mesh"
![image](https://user-images.githubusercontent.com/38707850/145345492-8fd33cf3-c166-4b81-b98f-e70c27603e47.png)
![image](https://user-images.githubusercontent.com/38707850/145345569-a53de58c-02ef-41fc-8671-5da5f028dc32.png)
